File: 0693-binary-number-with-alternating-bits/0693-binary-number-with-alternating-bits.go
func hasAlternatingBits(n int) bool {
    mask := 1
    for mask <= n {
        if ((mask & n) == 0) == (((mask<<1) & n) == 0) {
            return false
        } else {
            mask <<= 1
        }
    }
    return true;
}

File: 0693-binary-number-with-alternating-bits/0693-binary-number-with-alternating-bits.cpp
class Solution {
public:
    bool hasAlternatingBits(int n) {
        bool isOn = (1 & n) == 1;
        long long mask = 2;
        while(mask <= n) {
            if(((mask & n) != 0) == isOn) return false;
            mask <<= 1; isOn = !isOn;
        }
        return true;
    }
};